packages/sd-jwt/src/action-handler.ts

```diff
@@ -175,22 +190,37 @@ export class SDJwtPlugin implements IAgentPlugin {
   async verify(sdjwt: SDJwtVcInstance, context: IRequiredContext, data: string, signature: string) {
     const decodedVC = await sdjwt.decode(`${data}.${signature}`)
     const issuer: string = ((decodedVC.jwt as Jwt).payload as Record<string, unknown>).iss as string
-    if (!issuer.startsWith('did:')) {
-      throw new Error('invalid_issuer: issuer must be a did')
+    const header = (decodedVC.jwt as Jwt).header as Record<string, any>
+    const x5c: string[] | undefined = header?.x5c as string[]
+    let jwk: JWK | JsonWebKey | undefined = undefined
+    let key: IKey | undefined
+    if (issuer.includes('did:')) {
+      const identifier: IIdentifier = await context.agent.didManagerGet({ did: issuer.split('#')[0] })
+      key = await context.agent.getKey({ identifier })
+      if (key) {
+        const type = key?.type
+        const publicKey = key?.publicKeyHex
+        jwk = toJwk(publicKey, type)
+      }
     }
-    const didDoc = await context.agent.resolveDid({ didUrl: issuer })
-    if (!didDoc) {
-      throw new Error('invalid_issuer: issuer did not resolve to a did document')
+    if (x5c) {
+      const firstCert = x5c[0]
+      const certificateValidationResult: X509ValidationResult = await context.agent.verifyCertificateChain({
+        chain: [firstCert],
+        trustAnchors: [funkeTestCA, sphereonCA],
+      })
+
+      if (certificateValidationResult.error || !certificateValidationResult?.certificateChain) {
+        throw new Error('Certificate chain validation failed')
+      }
+      const pem = await x5cToPemCertChain([firstCert])
```

Comment:
The above verify method already returns the JWK for you for every certificate if there are no errors

Reply:
replaced it with getting the jwk value from the first element of the certificate chain validation result